File: IUST 6th Sem/Java/Work 2/StringPalindrome.java
import java.util.Scanner;

public class StringPalindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter String: ");
        String str = sc.nextLine();

        String rev = "";

        for(int i = str.length() -1; i >= 0; i--) {
            rev += str.charAt(i);
        }

        if(str.equals(rev))
            System.out.println("Palindrome");
        else  
            System.out.println("Not Palindrome");
    }
}

File: IUST 7th Sem/compilerDesign/StringPalindrome.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    string s;

    cout<< "Enter a String: ";
    cin >> s;

    string rev = s;

    int start = 0;
    int end = s.length() -1;

    while(start <= end){
        if(s[start] != s[end]){
            cout<<"String is not palindrome"<<endl;
            break;
        }
        start++;
        end--;
    }

    if(start > end) {
        cout<<"String is palindrome"<<endl;
    }
}